src/base/Context.ts

```diff
@@ -618,31 +623,40 @@ export class Context {
       })
     );
 
-    await P.map(claim1stKeys, async claim1stKey => {
-      const entries = await identity.claims.entries(claim1stKey);
-
-      entries.forEach(
-        ([
-          key,
-          { claim_issuer: claimissuer, issuance_date: issuanceDate, expiry: rawExpiry, claim },
-        ]) => {
-          const { target } = key.args[0] as Claim1stKey;
-          const expiry = rawExpiry ? momentToDate(rawExpiry.unwrap()) : null;
+    const trustedClaimIssuers = rawTrustedClaimIssuers?.map(trustedClaimIssuer =>
+      signerToString(trustedClaimIssuer)
+    );
 
-          if ((!includeExpired && (expiry === null || expiry > new Date())) || includeExpired) {
-            data.push({
-              target: new Identity({ did: identityIdToString(target) }, this),
-              issuer: new Identity({ did: identityIdToString(claimissuer) }, this),
-              issuedAt: momentToDate(issuanceDate),
-              expiry,
-              claim: meshClaimToClaim(claim),
-            });
+    const claimData = await P.map(claim1stKeys, async claim1stKey => {
+      const entries = await identity.claims.entries(claim1stKey);
+      const compactData = compact(
+        entries.map(
+          ([
+            key,
+            { claim_issuer: claimissuer, issuance_date: issuanceDate, expiry: rawExpiry, claim },
+          ]) => {
+            const { target } = key.args[0] as Claim1stKey;
+            const expiry = rawExpiry ? momentToDate(rawExpiry.unwrap()) : null;
+            if ((!includeExpired && (expiry === null || expiry > new Date())) || includeExpired) {
+              return {
+                target: new Identity({ did: identityIdToString(target) }, this),
+                issuer: new Identity({ did: identityIdToString(claimissuer) }, this),
+                issuedAt: momentToDate(issuanceDate),
+                expiry,
+                claim: meshClaimToClaim(claim),
+              };
+            }
+            return undefined;
           }
-        }
+        )
       );
+
+      return compactData;
     });
 
-    return data;
+    return flattenDeep(claimData).filter(({ issuer }) =>
+      trustedClaimIssuers ? trustedClaimIssuers.includes(issuer.did) : true
```

Comment:
`trustedClaimIssuers` can also contain Identities

Reply:
line 626:

```
    const trustedClaimIssuers = rawTrustedClaimIssuers?.map(trustedClaimIssuer =>
      signerToString(trustedClaimIssuer)
    );
```

Reply:
Missed that. Call the original variable `trustedClaimIssuers` and this one `claimIssuerDids`